pdp/handlers.go

```diff
@@ -909,15 +914,213 @@ func (p *PDPService) handleAddRootToProofSet(w http.ResponseWriter, r *http.Requ
+		SubrootOffset   int64  `db:"subroot_offset"`
+		SubrootSize     int64  `db:"subroot_size"`
+		AddMessageOK    *bool  `db:"add_message_ok"`
+		RootsAdded      bool   `db:"roots_added"`
+	}
+
+	var rootAdds []RootAddInfo
+	err = p.db.Select(ctx, &rootAdds, `
+		SELECT root, add_message_index, subroot, subroot_offset,
+		       subroot_size, add_message_ok, roots_added
+		FROM pdp_proofset_root_adds
+		WHERE proofset = $1 AND add_message_hash = $2
+		ORDER BY add_message_index, subroot_offset
+	`, proofSetID, txHash)
+	if err != nil {
+		http.Error(w, "Failed to query root additions: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if len(rootAdds) == 0 {
+		http.Error(w, "Root addition not found for given transaction", http.StatusNotFound)
+		return
+	}
+
+	// Step 5: Get transaction status from message_waits_eth
+	var txStatus string
+	err = p.db.QueryRow(ctx, `
+		SELECT tx_status FROM message_waits_eth WHERE signed_tx_hash = $1
+	`, txHash).Scan(&txStatus)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			http.Error(w, "Transaction status not found", http.StatusNotFound)
+			return
+		}
+		http.Error(w, "Failed to query transaction status: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Step 6: If transaction is confirmed and successful, get assigned root IDs
+	var confirmedRootIds []uint64
+	if txStatus == "confirmed" && len(rootAdds) > 0 && rootAdds[0].AddMessageOK != nil && *rootAdds[0].AddMessageOK {
+		// Get unique roots from the add transaction
+		uniqueRoots := make(map[string]bool)
+		for _, ra := range rootAdds {
+			uniqueRoots[ra.Root] = true
+		}
+
+		// Query pdp_proofset_roots for confirmed roots with their IDs
+		rootCids := make([]string, 0, len(uniqueRoots))
+		for root := range uniqueRoots {
+			rootCids = append(rootCids, root)
+		}
+
+		type ConfirmedRoot struct {
+			RootID uint64 `db:"root_id"`
+			Root   string `db:"root"`
+		}
+
+		var confirmedRoots []ConfirmedRoot
+		err = p.db.Select(ctx, &confirmedRoots, `
+			SELECT DISTINCT root_id, root
+			FROM pdp_proofset_roots
+			WHERE proofset = $1 AND root = ANY($2)
+			ORDER BY root_id
+		`, proofSetID, rootCids)
+		if err != nil {
+			log.Warnf("Failed to query confirmed roots: %v", err)
+			// Don't fail the request, just log the warning
+		} else {
+			// Extract just the root IDs
+			for _, cr := range confirmedRoots {
+				confirmedRootIds = append(confirmedRootIds, cr.RootID)
+			}
+		}
+	}
+
+	// Step 7: Build and send response
+	// Count unique roots for response
+	uniqueRootMap := make(map[string]bool)
+	for _, ra := range rootAdds {
+		uniqueRootMap[ra.Root] = true
+	}
+
+	// Check that all roots have the same RootsAdded value (consistency check)
+	if len(rootAdds) > 0 {
+		firstRootsAdded := rootAdds[0].RootsAdded
+		for _, ra := range rootAdds[1:] {
+			if ra.RootsAdded != firstRootsAdded {
+				http.Error(w, "Inconsistent rootsAdded state for this transaction's roots", http.StatusInternalServerError)
```

Comment:
I'm not so sure what value we get out of this.  It feels like we're testing the consistency model of yugabyte and I don't really care as a user for the purposes of this call.  I actually just care about the blockchain consistency model.

Reply:
Similar feeling about RootsAdded return value.  As user I care about chain and upload commitment and this seems to be some extraneous other bookkeeping.

Reply:
OK, your call, Aarsh and I went back and forth on this and I don't think I care. I'll remove the checks and `rootsAdded`.

Reply:
Oh, actually no I remember now - we _added_ this new field to the database, `roots_added`, it's the final check that tells us that Curio has fully processed it and it's onboarded. It's the direct equivalent of `proofset_created`. SO I need `roots_added`, it's just a question of whether we're OK trusting that the first row we get from the DB is representative of the whole lot. Perhaps there's a Curio-bug-reason that makes the first one `false` and the rest `true`; that's the logic that lead to all the looping and checking. So, your call on this but I'd really like the `roots_added` from for the SDK